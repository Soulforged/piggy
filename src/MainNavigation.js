//@flow

import React from 'react';
import { Image, Text } from 'react-native';
import { DrawerNavigator, StackNavigator } from 'react-navigation';

import styles from './Styles';

import HomeScreen from './HomeScreen';

const menuImg = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADMAAAAzCAYAAAA6oTAqAAAAEXRFWHRTb2Z0d2FyZQBwbmdjcnVzaEB1SfMAAABQSURBVGje7dSxCQBACARB+2/ab8BEeQNhFi6WSYzYLYudDQYGBgYGBgYGBgYGBgYGBgZmcvDqYGBgmhivGQYGBgYGBgYGBgYGBgYGBgbmQw+P/eMrC5UTVAAAAABJRU5ErkJggg==';

const Navigator = DrawerNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      drawerLabel: 'HomeScreen',
      drawerIcon: ({ rnNavTintColor }) => (
        <Image
          source={{ uri: menuImg }}
          style={[styles.icon, { tintColor: rnNavTintColor }]}
        />
      ),
    }
  },
});

const MainNavigation = StackNavigator(
  {
    Drawer: { screen: Navigator }
  },
  {
    headerMode: 'float',
    navigationOptions: ({ navigation }) => ({
      headerStyle: { backgroundColor: '#4C3E54' },
      title: 'Welcome!',
      headerTintColor: 'white',
      headerLeft: <Text onPress={() => navigation.navigate('DrawerToggle')}>Menu</Text>
    })
  }
);

export default MainNavigation;
