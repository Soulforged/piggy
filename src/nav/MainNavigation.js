//@flow

import React from 'react';
import { DrawerNavigator, StackNavigator } from 'react-navigation';
import Icon from 'react-native-vector-icons/Ionicons';

import { HomeScreen } from 'src/screens';

const Navigator = DrawerNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      drawerLabel: 'HomeScreen',
      drawerIcon: ({ rnNavTintColor }) => (
        <Icon name='md-menu' size={30} color={rnNavTintColor} />
      ),
    }
  },
});

const MainNavigation = StackNavigator(
  {
    Drawer: { screen: Navigator }
  },
  {
    headerMode: 'float',
    navigationOptions: ({ navigation }) => ({
      headerStyle: { backgroundColor: '#4C3E54' },
      title: 'Welcome!',
      headerTintColor: 'white',
      headerLeft: (
        <Icon.Button
          name='md-menu'
          size={30}
          color='white'
          onPress={() => navigation.navigate('DrawerToggle')}
        />
      )
    })
  }
);

export default MainNavigation;
