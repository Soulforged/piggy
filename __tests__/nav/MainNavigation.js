//@flow
import React from 'react';
import { shallow } from 'enzyme';

import MainNavigation from 'src/nav/MainNavigation';

it('renders without crashing', () => {
  const wrapper = shallow(<MainNavigation />);
  expect(wrapper).toMatchSnapshot();
});
