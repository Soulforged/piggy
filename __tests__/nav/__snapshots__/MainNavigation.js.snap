// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Navigator
  navigation={
    Object {
      "addListener": [Function],
      "dispatch": [Function],
      "getParam": [Function],
      "goBack": [Function],
      "navigate": [Function],
      "pop": [Function],
      "popToTop": [Function],
      "push": [Function],
      "replace": [Function],
      "setParams": [Function],
      "state": Object {
        "index": 0,
        "isTransitioning": false,
        "key": "StackRouterRoot",
        "routes": Array [
          Object {
            "index": 0,
            "isTransitioning": false,
            "key": "id-1482363367071-0",
            "routeName": "Drawer",
            "routes": Array [
              Object {
                "index": 0,
                "isTransitioning": false,
                "key": "DrawerClose",
                "params": undefined,
                "routeName": "DrawerClose",
                "routes": Array [
                  Object {
                    "key": "Home",
                    "params": undefined,
                    "routeName": "Home",
                  },
                ],
              },
              Object {
                "key": "DrawerOpen",
                "params": undefined,
                "routeName": "DrawerOpen",
              },
              Object {
                "key": "DrawerToggle",
                "params": undefined,
                "routeName": "DrawerToggle",
              },
            ],
          },
        ],
      },
    }
  }
/>
`;
